.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,37 @@
+name: Prepare Release
+
+# TODO: When v4 is ready to be merged this should be updated to trigger off main and support full releases via a workflow_dispatch event.
+# See https://github.com/rigetti/pyquil/issues/1501
+on:
+  push:
+    branches:
+      - v4
+  workflow_dispatch:
+    inputs:
+      type:
+        description: Bump versions and trigger a new release.
+        required: true
+        default: prerelease
+        options:
+          - prerelease
+
+jobs:
+  prepare-release:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.PAT }}
```

Comment:
Any particular reason this is here instead of on the `knope` step? I don't see it being used anywhere else. (Also for dry-run)